src/njoy21/input/DTFR/Card3a.hpp

```diff
@@ -0,0 +1,30 @@
+class Card3a {
+public:
+#include "njoy21/input/DTFR/Card3a/Mti.hpp"
+#include "njoy21/input/DTFR/Card3a/Mtc.hpp"
+#include "njoy21/input/DTFR/Card3a/Nlc.hpp"
+
+  Argument< Mti > mti;
+  Argument< Mtc > mtc;
+  Argument< Nlc > nlc;
+  
+  template< typename Char >
+  Card3a( iRecordStream< Char >& is )
+    try:
+      mti ( argument::extract< Mti >( is ) ),
+      mtc ( argument::extract< Mtc >( is ) ),
+      nlc ( argument::extract< Nlc >( is ) )
+      {
+        Card::clear(is);
+      }
+    catch( std::exception& e ){
+      Log::info("Trouble while validating Card3a");
+      throw e;
+    }
+
+    friend void swap( Card3a& left, Card3a& right ){
```

Comment:
This swap function was a workaround for a limitation in the implementation of `std::variant` we've been using. 

Jeremy and I bumped into another distinct limitation on Friday. Rather than continuing to kludge the code around those problems, I _swapped_ out our variant implementation for a more complete (and more thoroughly tested) one. I suspect you can safely remove this code after running the `fetch_subprojects` metaconfigure script from the high level directory of the source tree.